packages/cli/openapi-ir-to-fern/src/converters/convertEndpoint.ts

```diff
@@ -226,13 +228,136 @@ export function convertEndpoint({
+                : undefined,
+    };
+}
+
+function convertNamedFullExamplesToExampleTypeReferenceSchemas(
+    namedFullExamples: NamedFullExample[]
+): Record<string, RawSchemas.ExampleTypeReferenceSchema> {
+    const result: Record<string, RawSchemas.ExampleTypeReferenceSchema> = {};
+    namedFullExamples.map(
+        (namedFullExample) =>
+            (result[namedFullExample.name] = convertFullExampleToExampleTypeReferenceSchema(namedFullExample.value))
+    );
+    return result;
+}
+
+function convertFullExampleToExampleTypeReferenceSchema(
+    fullExample: FullExample
+): RawSchemas.ExampleTypeReferenceSchema {
+    return FullExample._visit<RawSchemas.ExampleTypeReferenceSchema>(fullExample, {
+        primitive: (primitive) => convertPrimitiveExampleToExampleTypeReferenceSchema(primitive),
+        object: (object) => convertFullExamplePropertiesToExampleTypeReferenceSchema(object.properties),
+        array: (array) => convertFullArrayExampleToExampleTypeReferenceSchema(array),
+        map: (map) => convertFullMapExampleToExampleTypeReferenceSchema(map),
+        oneOf: (oneOf) => convertFullOneOfExampleToExampleTypeReferenceSchema(oneOf),
+        enum: (value) => value,
+        literal: (value) => value,
+        unknown: (unknown) => convertFullExampleToExampleTypeReferenceSchema(unknown),
+        _unknown: () => "unknown",
+    });
+}
+
+function convertPrimitiveExampleToExampleTypeReferenceSchema(
+    primitiveExample: PrimitiveExample
+): RawSchemas.ExampleTypeReferenceSchema {
+    return PrimitiveExample._visit<RawSchemas.ExampleTypeReferenceSchema>(primitiveExample, {
+        int: (value: number) => value.toString(),
+        int64: (value: number) => value.toString(),
+        float: (value: number) => value.toString(),
+        double: (value: number) => value.toString(),
+        string: (value: string) => value,
+        datetime: (value: string) => value,
+        date: (value: string) => value,
+        base64: (value: string) => value,
+        boolean: (value: boolean) => value.toString(),
+        _unknown: () => "unknown",
+    });
+}
+
+function convertFullExamplePropertiesToExampleTypeReferenceSchema(
+    fullExampleProperties: Record<PropertyKey, FullExample>
+): RawSchemas.ExampleTypeReferenceSchema {
+    const properties: Record<string, RawSchemas.ExampleTypeReferenceSchema> = {};
+    Object.entries(fullExampleProperties).forEach(
+        ([propertyKey, fullExample]) =>
+            (properties[propertyKey] = convertFullExampleToExampleTypeReferenceSchema(fullExample))
+    );
+    return properties;
+}
+
+function convertFullArrayExampleToExampleTypeReferenceSchema(
+    fullExamples: FullExample[]
+): RawSchemas.ExampleTypeReferenceSchema {
+    return fullExamples.map((fullExample) => {
+        return convertFullExampleToExampleTypeReferenceSchema(fullExample);
+    });
+}
+
+function convertFullMapExampleToExampleTypeReferenceSchema(
```

Comment:
I think a better method name for all of these is just `convertMapExampleToFern` or something

Reply:
`convertFullMapExampleToExampleTypeReferenceSchema` is innacurate because you aren't converting the thing to an "exampleTypeReferenceSchema". You're converting it to an example, and the example has to be typed as that schema

Reply:
Totally up for renaming, but I chose `ExampleTypeReferenceSchema` because that's actually the name of the type we're returning from all these cases - you can see it in the signature. But I agree it's still not very clear, I like your suggestion better so let's go with that.

Reply:
maybe `convertMapToFernExample`? 🤷 not too opinionated but i think something shorter would be nice

Reply:
yup the type here is a zod schema, and zod helps us do validation of the raw yaml. `ExampleTypeReferenceSchema` makes sense as a name for the zod schema, but its a little confusing for this method cause we aren't returning a schema we are returning the actual example itself.